src/ra_log_segment_writer.erl

```diff
@@ -119,8 +119,25 @@ segments_for(UId, #state{data_dir = DataDir}) ->
     SegFiles = lists:sort(filelib:wildcard(filename:join(Dir, "*.segment"))),
     SegFiles.
 
-handle_cast({mem_tables, Tables, WalFile}, State0) ->
-    State = lists:foldl(fun do_segment/2, State0, Tables),
+handle_cast({mem_tables, Tables, WalFile}, State) ->
+
+    Degree = erlang:system_info(schedulers),
+    _ = [begin
+             {_, Failures} = ra_lib:partition_parallel(
+                               fun (E) ->
+                                       ok = do_segment(E, State),
+                                       true
+                               end, Tabs, infinity),
+             case Failures of
+                 [] ->
+                     %% this is what we expect
+                     ok;
+                 _ ->
+                     ?ERROR("segment_writer: ~b failures encounted during segment"
+                            " flush Errors ~P", [length(Failures), Failures, 32]),
```

Comment:
`~P` is for precision or `~p` with debt. It would fail to format if not used with a depth. Did you mean `~p`?

Reply:
Curiously this does work as expected.